src/kerberos.c

```diff
@@ -211,14 +211,56 @@ static PyObject *authGSSClientClean(PyObject *self, PyObject *args)
     return Py_BuildValue("i", result);
 }
 
-static PyObject *authGSSClientStep(PyObject *self, PyObject *args)
+static PyObject *buildChannelBindingsStruct(PyObject *self, PyObject *args, PyObject* keywds)
+{
+    int initiator_addrtype = GSS_C_AF_UNSPEC;
+    int acceptor_addrtype = GSS_C_AF_UNSPEC;
+    int initiator_length = 0;
+    int acceptor_length = 0;
+    int application_length = 0;
+    const char *initiator_address = NULL;
+    const char *acceptor_address = NULL;
+    const char *application_data = NULL;
+    int result = 0;
+
+    PyObject *pychan_bindings = NULL;
+    struct gss_channel_bindings_struct *input_chan_bindings;
+	static char *kwlist[] = {"initiator_addrtype", "initiator_address", "accept_addrtype",
+	                        "acceptor_address", "application_data", NULL};
+
+	if (!PyArg_ParseTupleAndKeywords(args, keywds, "|is#is#s#", kwlist,
```

Comment:
Since the PyCObject / PyCapsule used to hold the the binding struct may outlive the Python strings used to pass in the addresses, I think you need to use es# here, and call PyMem_Free on each string in your destructor.